sql/core/src/main/java/org/apache/spark/sql/sources/v2/SupportsRead.java

```diff
@@ -25,7 +25,7 @@
  * {@link #newScanBuilder(DataSourceOptions)} that is used to create a scan for batch, micro-batch,
  * or continuous processing.
  */
-interface SupportsRead extends Table {
+public interface SupportsRead extends Table {
```

Comment:
shall we remove this interface as well? We can move `newScanBuilder` to table and throw exception by default. Tables  that reports batch/stream scan capability should overwrite `newScanBuilder`

Reply:
I like having this because it maintains separation between the read/write API and the catalog API. We could update the read and write API later or add a new one by adding a different read trait, without changing how catalogs and tables work. So I think it is worth keeping `SupportsRead` and `SupportsWrite`.